File: 0151-reverse-words-in-a-string/0151-reverse-words-in-a-string.py
class Solution:
    def reverseWords(self, s: str) -> str:
        s = s.split()
        
        start = 0
        end = len(s) - 1
        
        while start < end:
            s[start], s[end] = s[end], s[start]
            start += 1
            end -= 1
        return " ".join(s)

File: 0151-reverse-words-in-a-string/0151-reverse-words-in-a-string.java
class Solution {
    public String reverseWords(String s) {
        String[] splittedWOrds = s.trim().split("\\s+"); 
        System.out.println(Arrays.toString(splittedWOrds)); 
        int i = 0, j = splittedWOrds.length - 1; 
        while( i < j) {
            String temp = splittedWOrds[i]; 
            splittedWOrds[i] = splittedWOrds[j]; 
            splittedWOrds[j] = temp; 
            i++; 
            j--; 
        }        
        return String.join(" ", splittedWOrds);
    }
}